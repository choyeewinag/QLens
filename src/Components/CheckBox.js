import React, { useState, useEffect } from 'react';
// import MongoSchemaIDE from 'MongoSchemaIDE';

const CheckBox = (props) => {
  return (
    <div className="checkboxContainer" id="box">
      <form className="checkboxForm">
        <label className="checkboxLabel">
          <input type="checkbox" name={props.name} onClick={props.clicked} />
          {/* name is the key which is the obj */}
          {props.name}
        </label>
      </form>
    </div>
  )
}



export default CheckBox;