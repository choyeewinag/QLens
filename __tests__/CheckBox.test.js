import React, { useState, useEffect } from 'react';
import renderer from 'react-test-renderer';
import CheckBox from '../src/Components/CheckBox';
import ShallowRenderer from 'react-test-renderer/shallow';

//Check if testing works
// describe('My Test Suite', () => {
//   it('My Test Case', () => {
//       expect(true).toEqual(true);
//   });
//   });

//snapshots

describe('CheckBox Component Renders Correctly', () => {
test('snapshot of CheckBox Component renders', () => {
    const component = renderer.create(<CheckBox/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
});

// import TestRenderer from 'react-test-renderer';
// import ShallowRenderer from 'react-test-renderer/shallow';


// Basic Test with React-test-renderer
// it('renders correctly react-test-renderer', () => {
//   const renderer = new ShallowRenderer();
//   renderer.render(<Basic />);
//   const result = renderer.getRenderOutput();
//
//   expect(result).toMatchSnapshot();
// });

