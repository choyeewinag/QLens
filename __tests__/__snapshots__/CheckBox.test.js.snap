// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox Component Renders Correctly snapshot of CheckBox Component renders 1`] = `
<div
  className="checkboxContainer"
  id="box"
>
  <form
    className="checkboxForm"
  >
    <label
      className="checkboxLabel"
    >
      <input
        type="checkbox"
      />
    </label>
  </form>
</div>
`;
